[Elizabeth Chow Ting San] iP

lizchow opened the pull request:
### Duke Pro
> "Your mind is for having ideas, not holding them." – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things yo need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER FAST* to use

All you need to do is,
1. download it from [here](https://github.com/lizchow/ip/releases/download/A-Jar/ip.jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

***
If you are a Java Programmer, you can use it to practice Java too. Here's the `main`method:
```java
public class Duke {
    public static void main(String[] args) {
        new Duke().run();
    }
}
```

Commit:
Add Gradle support

lizchow committed:
Add Level 1

lizchow committed:
Add Level-2

lizchow committed:
Add Level-3

lizchow committed:
extract out count and tasks from Duke main

lizchow committed:
Add Level-4

lizchow committed:
Add A-TextUiTesting

lizchow committed:
Add Level-5

lizchow committed:
Add Level-6

lizchow committed:
Add A-Enums

lizchow committed:
refactor to better handle empty tasks in done, delete

lizchow committed:
abstract out greet from main()

lizchow committed:
Add Level-7

lizchow committed:
Add Level-8

lizchow committed:
Merge branch 'branch-Level-7'

lizchow committed:
Merge branch 'branch-Level-8'

lizchow committed:
Fix datetime parsing for Deadline object

lizchow committed:
Add A-MoreOOP

lizchow committed:
Add A-Packages

lizchow committed:
Add A-JUnit

lizchow committed:
Add A-JavaDoc

lizchow committed:
Add A-CodingStandard

lizchow committed:
Add Level-9

lizchow committed:
Merge branch 'branch-A-JavaDoc'

lizchow committed:
Merge branch 'branch-A-CodingStandard'

lizchow committed:
Merge branch 'branch-Level-9'

lizchow committed:
Fix formatting

lizchow committed:
Merge branch 'add-gradle-support' of github.com:lizchow/ip into branch-A-Gradle

lizchow committed:
Upgrade to JUnit5 from JUnit 4.13

lizchow committed:
Add checkstyle xml files

lizchow committed:
Fix checkstyle errors

lizchow committed:
Merge branch 'branch-A-CheckStyle'

lizchow committed:
Complete javafx tutorial1

lizchow committed:
Complete javafx tutorial2

lizchow committed:
Complete javafx tutorial 3

lizchow committed:
Complete javafx tutorial 4

lizchow committed:
Respond to user input with the duke response

lizchow committed:
Fix checkstyle errors

lizchow committed:
Merge branch 'branch-Level-10'

lizchow committed:
Add A-Varargs

lizchow committed:
Merge branch 'branch-A-Varargs'

yzhilong commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,60 @@
+     *
+     * @return {@code true} if the program should exit, {@code false} otherwise.
+     */
+    public abstract boolean isExit();
```
```suggestion
    public boolean isExit() {
        return false;
    }
```

Would it be better if the logic for isExit is handled within the command class? Since every other command class (excluding ExitCommand) returns false.

lizchow replied:
Yep, I think it makes more sense to be handled in the command class!

yzhilong commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,129 @@
+     * Writes to file at the specified file path with the given tasks.
+     *
+     * @param tasks Tasks to be written to be file.
+     *
```
```suggestion
```
I don't think that you need to add a new line between `@param` and `@throws` here

lizchow replied:
Hmm ok

yzhilong commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,118 @@
+            int taskNo = Integer.parseInt(input) - 1;
+            return new DoneCommand(String.valueOf(taskNo));
+        } catch (NumberFormatException e) {
+            throw new DukeException("Enter a number for a done action!");
```
Would it make the error clearer to the user if you had different exception classes extending DukeException?

lizchow replied:
haha yes

yzhilong commented on src/test/java/duke/TaskListTest.java:
```diff
@@ -0,0 +1,114 @@
+        TaskList taskList = new TaskList(new ArrayList<Task>());
+        Assertions.assertThrows(DukeException.class, () -> taskList.markTaskDone(2));
+    }
+}
```
Very comprehensive testing!! 👍

yzhilong reviewed (commented):
LGTM! I like how comprehensive your javadocs are!

lizchow committed:
Add assertions

Use assert feature to assert that when commands are executed,
it contains the correct number of arguments.

Asserting this command will ensure that the Command classes are
not abused during development.

lizchow committed:
Improve code quality

lizchow committed:
Use Streams

lizchow committed:
Merge pull request #1 from lizchow/branch-A-Assertions

Add assertions

lizchow committed:
Merge pull request #2 from lizchow/branch-A-CodeQuality

Improve code quality

lizchow committed:
Merge pull request #3 from lizchow/branch-A-Streams

Use Streams

lizchow committed:
Add update functionality for Duke

ALlows user to update the tasks with the following command format:
1. `update [taskNo] [task title] /date [date]`
2. `update [taskNo] [task title]`
3. `update [taskNo] /date [date]`

Displays error message if the input command is not in the given formats.

lizchow committed:
Merge branch 'branch-C-Update'

lizchow committed:
Add CI

lizchow committed:
Fix CI

lizchow committed:
Add A-BetterGui

lizchow committed:
remove text-ui-test

lizchow committed:
Add ui screenshot

lizchow committed:
Update application main class

lizchow committed:
Change font to comic sans

lizchow committed:
Change background color to label only

lizchow committed:
Add A-UserGuide

lizchow committed:
Fix checkstyle

lizchow committed:
Update README.md

lizchow committed:
Fix comments

lizchow committed:
Merge branch 'master' of github.com:lizchow/ip

lizchow committed:
Update README.md

lizchow committed:
Update README.md

lizchow committed:
Update README.md
